packages/core/src/external/fhir/shared/bundle.ts

```diff
@@ -157,19 +158,41 @@ export function buildSearchSetBundle<T extends Resource = Resource>({
   return buildBundle({ type: "searchset", entries }) as SearchSetBundle<T>;
 }
 
+/**
+ * @deprecated - use createBundleEntry instead
+ */
 export const buildBundleEntry = <T extends Resource>(resource: T): BundleEntry<T> => {
   const fullUrl = buildFullUrl(resource);
   return {
     ...(fullUrl ? { fullUrl } : {}),
     resource,
   };
 };
-export const buildFullUrl = <T extends Resource>(resource: T): string | undefined => {
+
+export function createBundleEntry(entry: BundleEntry): BundleEntry {
```

Comment:
In theory, we should get the bundle as a reference so we know the type and can include the request or not based on the type - [link](https://www.hl7.org/fhir/R4/bundle-definitions.html#Bundle.entry:~:text=to%20this%20element.-,Bundle.entry.request,-Element%20Id):

> entry.request mandatory for batch/transaction/history, otherwise prohibited

Something like:

```ts
export function buildCreateBundleEntry(bundle: Bundle) {
  return (entry: BundleEntry): BundleEntry {
    // only add the request if the bundle.type is one of batch/transaction/history
  }
}
```
...and then use like so:
```ts
const createBundleEntry = buildCreateBundleEntry(updatedBundle);
const fullBundleEntries = updatedBundle.entry.map(createBundleEntry);
```